app/core/engine.py

```diff
@@ -189,6 +194,44 @@ async def initialize_agent(aid):
             cdp_tools = [tool for tool in cdp_tools if tool.name in agent.cdp_skills]
         tools.extend(cdp_tools)
 
+    if agent.goat_enabled:
+        crossmint_wallet, smart_wallet_data = init_smart_wallet(
+            config.crossmint_api_key,
+            "base",
+            config.rpc_crossmint,
+            config.ens_crossmint,
+            (
+                agent.crossmint_wallet_data.get("smart")
+                if agent.crossmint_wallet_data
+                else None
+            ),
+        )
+
+        # save the wallet after first create
+        if (
+            not agent.crossmint_wallet_data
```

Comment:
not add to agent, add to agent data.  agent are pure config by user.

Reply:
set_data is right, it's in agent_data, but this check is wrong.

Reply:
I followed CDP, Do you mean we no longer store the wallet data in the agent?

Reply:
Yes, that's because backward compatibility hasn't been removed yet, and it should be removed now.

Currently, the user-maintained config is stored in the agent table, while the mutable data we generate is placed in agent_data.